src/QldbDriver.ts

```diff
@@ -156,36 +169,95 @@ export class QldbDriver {
      * @throws {@linkcode InvalidSessionException} When a session expires either due to a long running transaction or session being idle for long time.
      * @throws {@linkcode BadRequestException} When Amazon QLDB is not able to execute a query or transaction.
      */
-    async executeLambda(
-        transactionLambda: (transactionExecutor: TransactionExecutor) => any,
+    async executeLambda<Type>(
+        transactionLambda: (transactionExecutor: TransactionExecutor) => Promise<Type>,
         retryConfig?: RetryConfig
-    ): Promise<any> {
-        let session: QldbSession = null;
-        retryConfig = (retryConfig == null) ? this._retryConfig : retryConfig;
-        const transactionExecutionContext: TransactionExecutionContext = new TransactionExecutionContext();
-        let transactionExecutionAttempt: number = 0;
-        while(true) {
-            try  {
-                transactionExecutionAttempt += 1;
-                session = await this.getSession();
-                return await session.executeLambda(transactionLambda, retryConfig, transactionExecutionContext);
-            } catch(err) {
-                /* This is a guard condition to prevent the driver from entering an infinite loop
-                if all the sessions start resulting in InvalidSessionException
-                */
-                if (transactionExecutionAttempt >= this._maxConcurrentTransactions + 3) {
-                    throw err;
+    ): Promise<Type> {
+        if (this._isClosed) {
+            throw new DriverClosedError();
+        }
+
+        // Acquire semaphore and get a session from the pool
+        const getSession: (thisDriver: QldbDriver, startNewSession: boolean) => Promise<QldbSession> = 
```

Comment:
This extra typing information is probably unnecessary/can be inferred.

Reply:
Done.